beacon-chain/blockchain/core.go

```diff
@@ -51,18 +47,22 @@ func NewBeaconChain(db ethdb.Database) (*BeaconChain, error) {
 		db:    db,
 		state: &beaconState{},
 	}
-	hasActive, err := db.Has([]byte(activeStateLookupKey))
+	hasCrystallized, err := db.Has(CrystallizedStateLookupKey)
 	if err != nil {
 		return nil, err
 	}
-	hasCrystallized, err := db.Has([]byte(crystallizedStateLookupKey))
+	hasGenesis, err := db.Has(GenesisLookupKey)
 	if err != nil {
 		return nil, err
 	}
-	hasGenesis, err := db.Has([]byte("genesis"))
+
+	active, _, err := types.NewGenesisStates()
```

Comment:
Hmm what's going on here. Why is the active state being set to the genesis state, and why is `types.NewGenesisState()` being called twice?

Reply:
the active state is recomputed whenever we run the chain service , but the service needs an active state at startup otherwise it fails. I will remove the additional call for the genesis state